### PR TITLE
LPS-169416 New User Segment sidebar trigger switch is not correctly controlled

### PR DESCRIPTION
**Pull request by Stephy6**
Motivation
=======
The motivation behind this ticket is to solve an accessibility issue related to the Segments Editor. Is about the toggle button which is used to change between the edit and view mode of the Segments Editor and opens/closes the Segments Contributors Panel. This button isn't accessible since it doesn't inform the user with the proper informative messages and the button is not related to the panel it opens.
[LPS-169416](https://issues.liferay.com/browse/LPS-169416)

Proposed Solution
========
To solve the problem we change the button icon and add the labels, titles and roles necessary. Also we use the focus method combined with tabIndex=-1 attribute to handle the focus when the panel is opened. 

Steps to Verify
========
1. Go to People -> Segments
2. Create a Segments by clicking on (+)
3. Activate ScreenReader and navigate by the keyboard to the configuration button
4. Click enter on the button and check that the focus is changed to the opened panel
5. Listen to the ScreenReader and check that informs of the two possible messages of the button (Enter View/Edit Mode) and says the panel name (Segments Contributors Panel) correctly

Screenshot 2023-03-28 at 19 11 08

### CODE REVIEW

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js [hunk 1]
```diff
@@ -502,25 +501,7 @@ class SegmentEdit extends Component {
...
-											onChange={this._handleCriteriaEdit}
-											title={Liferay.Language.get(
-												'edit-segment'
-											)}
-										/>
-									</div>
-								</div>
-
-								<div className="btn-group">
+								<div className="btn-group mr-3">
```
Comment (sandrodw3) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js:512]: I think we can remove the `mr-3`, right?

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js [hunk 2]
```diff
@@ -546,6 +527,23 @@ class SegmentEdit extends Component {
...
 										</ClayButton>
 									</div>
+
+									<div className="btn-group-item">
+										<ClayButtonWithIcon
+											aria-label={Liferay.Language.get(
+												'edit-segment'
+											)}
+											borderless={true}
+											className="mr-3"
```
Comment (sandrodw3) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js:543]: Same here, I think we can remove this one

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js [hunk 3]
```diff
@@ -465,6 +469,10 @@ class SegmentEdit extends Component {
 		const showDisabledSegmentationAlert =
 			!isSegmentationEnabled && !isSegmentationDisabledAlertDismissed;
 
+		const title = editing
```
Comment (sandrodw3) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js:472]: Maybe we could give this variable a more specific name, like `editButtonTitle` or something like that 😄

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js [hunk 4]
```diff
@@ -530,18 +538,16 @@ class SegmentEdit extends Component {
...
-												'edit-segment'
-											)}
+											aria-label={title}
 											borderless={true}
 											className="mr-3"
 											displayType="secondary"
 											onClick={this._handleCriteriaEdit}
 											outline={true}
+											role="tab"
+											small={true}
```
Comment (sandrodw3) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js:548]: I think we should use here `size="sm"` as `small` is deprecated in `ClayButton`

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/css/_contributor_builder.scss [hunk 5]
```diff
@@ -85,7 +85,6 @@
 		position: fixed;
 		right: 0;
 		top: 121px;
-		visibility: hidden;
```
Comment (sandrodw3) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/css/_contributor_builder.scss:88]: Fantastic catch here 🎉 I would give a bit more of detail in the commit message no why this is not needed, something like "Remove visibility rule as the width one is enough"

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties
Comment (sandrodw3) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:1]: I think changes in this file should be in the commit in which you use the keys for the first time, and the buildLang commit should also include the changes generated by the buildLang command

**Review by sandrodw3 — COMMENTED** [review]
This works super good @Stephy6! 😄 nice catch with the visibility issue!

I added some tiny comments with suggestions for some improvements, please take a look!

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js (hunk 1 above)
Comment (Stephy6) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js:512]: Done, thank you!

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/css/_contributor_builder.scss (hunk 5 above)
Comment (Stephy6) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/css/_contributor_builder.scss:88]: True, thanks!

**Comment by sandrodw3** [comment]
Thanks @Stephy6, good job! 😄

**Comment by sandrodw3** [comment]
ci:forward

**Review by sandrodw3 — APPROVED** [review]

**Comment by sandrodw3** [comment]
ci:forward

**Comment by jkappler** [comment]
ci:forward

**Comment by Stephy6** [comment]
@jkappler I've checked the test and It is related with the PR and I think it is about the new button we're using here. Before we were using a toggle button which had a checked and an unchecked state and now we're using a cnf button which changes the title (the pop-up message) when it changes the state.

**Comment by Tim-Cao** [comment]
See https://github.com/liferay-echo/liferay-portal/pull/11631

**Pull request closed without merging**